xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -786,16 +791,21 @@ $fn
 #macro(panelhiddenheader $title)
 ## Note: We pass the Panel name as an HTML class attribute so that it's possible to style
 ## the Panels selectively using CSS.
-#set ($specialClassAttribute = '')
-#if ($paneldoc && $paneldoc != '')
-  #set ($specialClassAttribute = $util.convertToAlphaNumeric($paneldoc.name))
+#set ($defaultDocumentName = $services.model.getEntityReference('DOCUMENT', 'default').name)
+#if ($paneldoc.documentReference.name == $defaultDocumentName)
+  #set ($serializedDocument = $!services.model.serialize($paneldoc.documentReference.parent, 'compactwiki'))
+#else
+  #set ($serializedDocument = $!services.model.serialize($paneldoc.documentReference, 'compactwiki'))
 #end
+#set ($specialClassAttribute = $stringtool.toAlphaNumeric("$!serializedDocument"))
+## We keep this class for compatibility purpose but the previous one should be used instead.
```

Comment:
This loos like a duplicate of what's above. I'd suggest introducing an internal velocity macro for that.

Reply:
Or even a `#define` since the duplication seems to go deep even in the generated wiki syntax.